src/rgw/setup/rgw.py

```diff
@@ -109,7 +109,7 @@ def config(conf: MappedConf):
         # Update motr fid,endpoint config in cortx_rgw.conf.
         instance = 1
         while instance <= client_instance_count:
-            service_endpoints = Rgw._parse_endpoint_values(conf)
+            service_endpoints = Rgw._parse_endpoint_values(conf, instance, client_instance_count)
```

Comment:
so 'instance' is hardcoded to 1 anyways today, because of the lack of corresponding key in GConf ? And when thats added we will have to update the code to read instance from that config !
If yes, can we read this value from old key itself, and in absence of which it will default value as 1, instead of totally hardcoding ?

Reply:
No, we don't need to update our code, 
We are reading num_client_instance for rgw_s3 service**('cortx>motr>clients[%s]>num_instances'**) from Gconf in [_get_num_client_instances](https://github.com/Seagate/cortx-rgw-integration/blob/37e741a572b7fc41d0a8d4c333e933031ff2bcc3/src/rgw/setup/rgw.py#L395) function.
 ```
  while conf.get(const.CLIENT_INSTANCE_NAME_KEY % client_idx) is not None:
            name = Rgw._get_cortx_conf(conf, const.CLIENT_INSTANCE_NAME_KEY % client_idx)
            if name == const.COMPONENT_SVC_NAME:
                num_instances = int(Rgw._get_cortx_conf(conf, const.CLIENT_INSTANCE_NUMBER_KEY % client_idx))
                break
            client_idx = client_idx + 1
```
Here, if that key is not present for the service name, then it is returning the default value 1.
Now as we have service_name = **rgw_s3** and in Gconf it is **rgw**, check **( if name == const.COMPONENT_SVC_NAME:)**
is failing and we are proceeding with default value, but once they change the service name, we can get the actual num_instance value for **rgw_s3** and our code will work as it is, No code changes will be needed from our side.

Reply:
great, thanks for explaining